### PR TITLE
type-connect part 7: settings, store, stories, util

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/store/configure-store.js [hunk 1]
```diff
@@ -91,7 +90,6 @@ const errorCatching = store => next => action => {
 const middlewares = [
   errorCatching,
   createSagaMiddleware(crashHandler),
-  thunkMiddleware,
```
Comment (chrisnojima) [inline shared/store/configure-store.js]: the original impetus for this, remove thunk

File: shared/settings/about-container.js [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 import About from './about'
 import {HeaderHoc} from '../common-adapters'
-import {connect} from 'react-redux'
 import {version} from '../constants/platform'
-import {defaultProps, compose} from 'recompose'
 
-import type {Dispatch} from '../constants/types/flux'
-
-const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, navigateAppend}) => ({
+const mapStateToProps = () => ({version})
```
Comment (songgao) [inline shared/settings/about-container.js]: Is there a reason not to just put this in `mapDispatchToProps` and leave this omitted?

File: shared/settings/about-container.js [hunk 3]
```diff
@@ -27,8 +25,9 @@ const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, navigateAppend}) =>
   title: 'About',
 })
 
-const connectedHeaderHoc = compose(connect(null, mapDispatchToProps), HeaderHoc, defaultProps({version}))(
-  About
-)
+const connectedHeaderHoc = Container.compose(
+  Container.connect(mapStateToProps, mapDispatchToProps, (s, d, o) => ({...s, ...d, ...o})),
```
Comment (songgao) [inline shared/settings/about-container.js]: Hmm is `mergeProps` required now?

Comment (chrisnojima) [reply shared/settings/about-container.js]: yes it is

File: shared/settings/about-container.js (hunk 2 above)
Comment (chrisnojima) [reply shared/settings/about-container.js]: mapDispatchToProps should only be creating action props, not state props, this kind of thing belongs in MSP or mergePRops

**Pull request merged**